metadata-service/auth-impl/src/main/java/com/datahub/telemetry/TrackingService.java

```diff
@@ -0,0 +1,168 @@
+      ImmutableList.of(EVENT_TYPE_FIELD, SIGN_UP_TITLE_FIELD, QUERY_FIELD, ENTITY_TYPE_FIELD, ENTITY_TYPE_FILTER_FIELD,
+          PAGE_NUMBER_FIELD, PAGE_FIELD, TOTAL_FIELD, INDEX_FIELD, RESULT_TYPE_FIELD, RENDER_ID_FIELD, MODULE_ID_FIELD,
+          RENDER_TYPE_FIELD, SCENARIO_TYPE_FIELD));
+
+  private static final String ACTOR_URN_FIELD = "actorUrn";
+  private static final String ORIGIN_FIELD = "origin";
+  private static final String ENTITY_URN_FIELD = "entityUrn";
+  private static final String ENTITY_URNS_FIELD = "entityUrns";
+  private static final String GROUP_NAME_FIELD = "groupName";
+  private static final String SECTION_FIELD = "section";
+  private static final String ENTITY_PAGE_FILTER_FIELD = "entityPageFilter";
+  private static final String PATH_FIELD = "path";
+  private static final Set<String> ALLOWED_OBFUSCATED_EVENT_FIELDS = new HashSet<>(
+      ImmutableList.of(ACTOR_URN_FIELD, ORIGIN_FIELD, ENTITY_URN_FIELD, ENTITY_URNS_FIELD, GROUP_NAME_FIELD,
+          SECTION_FIELD, ENTITY_PAGE_FILTER_FIELD, PATH_FIELD));
+
+  private final MixpanelAPI _mixpanelAPI;
+  private final MessageBuilder _mixpanelMessageBuilder;
+  private final SecretService _secretService;
+  private final EntityService _entityService;
+  private final GitVersion _gitVersion;
+  private final ObjectMapper _objectMapper = new ObjectMapper();
+  private String _clientId;
+
+  public void emitAnalyticsEvent(@Nonnull final JsonNode event) {
+    final JSONObject sanitizedEvent = sanitizeEvent(event);
+    final String sanitizedActor;
+    final String eventType;
+    try {
+      sanitizedActor = sanitizedEvent.getString(ACTOR_URN_FIELD);
+      eventType = sanitizedEvent.getString(EVENT_TYPE_FIELD);
+    } catch (JSONException e) {
+      throw new RuntimeException("Failed to parse actor and event type from event", e);
+    }
+
+    String eventId = String.join(":", getClientId(), sanitizedActor);
+
+    try {
+      _mixpanelAPI.sendMessage(_mixpanelMessageBuilder.event(eventId, eventType, sanitizedEvent));
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to send event to Mixpanel", e);
+    }
+  }
+
+  @Nonnull
+  public String getClientId() {
+    // Return cached client id if it exists
+    if (_clientId != null) {
+      return _clientId;
+    }
+
+    Urn clientIdUrn = UrnUtils.getUrn(CLIENT_ID_URN);
+    // Create a new client id if it doesn't exist
+    if (!_entityService.exists(clientIdUrn)) {
+      return createClientIdIfNotPresent(_entityService);
+    }
+
+    // Otherwise, return the existing client id from the metadata store
+    RecordTemplate clientIdTemplate = _entityService.getLatestAspect(clientIdUrn, CLIENT_ID_ASPECT);
+    // Should always be present here from above, so no need for null check
+    _clientId = ((TelemetryClientId) clientIdTemplate).getClientId();
+    return _clientId;
+  }
+
+  @Nonnull
+  JSONObject sanitizeEvent(@Nonnull final JsonNode event) {
+    final JSONObject eventObj;
```

Comment:
for the sake of no confusion, would be good to call this `unsanitizedEventObj`

Reply:
Good call, renamed